### PR TITLE
Document some distortion world and party functions

### PR DESCRIPTION
**Pull request by ravepossum**
Follow up to PR #207 

This picks off and documents a few functions and constants that are referenced in `field_control.c`, mostly related to the party and the distortion world.

### CODE REVIEW

File: src/player_avatar.c
Comment (lhearachel) [inline src/player_avatar.c:897]: Does this match?

```c
return (state == AVATAR_DISTORTION_STATE_NONE || state == AVATAR_DISTORTION_STATE_ACTIVE) == FALSE;
```

Or, if possible:

```c
return state != AVATAR_DISTORTION_STATE_NONE && state != AVATAR_DISTORTION_STATE_ACTIVE;
```

I'd recommend looking at the ASM here and fiddling with the branch-return sequence to see if you can find something that matches more cleanly.

File: src/player_avatar.c
Comment (lhearachel) [inline src/player_avatar.c:904]:
```c
return state == AVATAR_DISTORTION_STATE_NONE
    || state == AVATAR_DISTORTION_STATE_ACTIVE
    || state == AVATAR_DISTORTION_STATE_FLOOR;
```

File: src/unk_02052C6C.c
Comment (lhearachel) [inline src/unk_02052C6C.c:212]: [aside] Oh, hey, hall-of-fame code.

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:21]: How did you try to simplify this?

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:90]:
```suggestion
        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) != FALSE) {
```

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:145]:
```suggestion
        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) == FALSE) {
```

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:133]:
```suggestion
    GF_ASSERT(FALSE);
```

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:120]: I feel like the naming between this and `Party_FirstMon` is a little confusing; maybe:

- `Party_FirstBattler` -> `Party_FindFirstEligibleBattler`
- `Party_FirstMon` -> `Party_FindFirstHatchedMon`

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:161]:
```suggestion
    u8 championRibbon = TRUE;
```

File: src/unk_02054884.c
Comment (lhearachel) [inline src/unk_02054884.c:158]:
```suggestion
void Party_GiveChampionRibbons (Party *party)
```

My logic here is that `SetChampionRibbons` implies to me that I can specify whether the members have the ribbon or not. `Give` is unambiguous; after this routine, all party members will have the ribbon flag flipped to `TRUE`.

File: src/unk_0205F180.c
Comment (lhearachel) [inline src/unk_0205F180.c:871]: Unnecessary scope block?

**Review by lhearachel — COMMENTED** [review]
As a general comment leading out of discussion on Discord, we're considering moving towards prefixing subroutines with appropriate verbage as a general naming scheme. Feel free to hold off on that until we reach a full consensus on the approach, but I wanted to make sure that I recorded it here.

File: src/unk_02054884.c
Comment (ravepossum) [reply src/unk_02054884.c:161]: The fact that this doesn't match with a `BOOL` instead of `u8` makes me sad.

Comment (lhearachel) [reply src/unk_02054884.c:161]: Memory sizes haha!

We should still make use of the `BOOL` constants where possible, even if the underlying type is not actually `BOOL` (read: `int`).

File: src/unk_02054884.c
Comment (ravepossum) [reply src/unk_02054884.c:21]: I tried these:
```
return !(Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0) || !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
```

```
return Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) != 0 || !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
```
```
return !(Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0 && Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL));
```

File: src/player_avatar.c
Comment (lhearachel) [reply src/player_avatar.c:897]: Did some fiddling and we determined that the feasible match here is:

```c
return PlayerAvatar_DistortionState(playerAvatar) > (u32)AVATAR_DISTORTION_STATE_FLOOR;
```

That's pretty smelly, so we'll keep this sentinel structure instead.

File: src/unk_02054884.c
Comment (lhearachel) [reply src/unk_02054884.c:21]: I took a stab at this and got to this:

```c
BOOL Pokemon_CanBattle(void *mon)
{
    return Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) != 0
        && !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
}
```

which is logically (and functionally) equivalent, but does not match byte-for-byte. So, meh. Keep this as-is.

File: src/player_avatar.c
Comment (lhearachel) [reply src/player_avatar.c:904]: Does it really not match with this change? Also, why remove this indentation in the most recent revision?

Comment (ravepossum) [reply src/player_avatar.c]: My monkey brain read right past the `return` in your comment and read it as a suggestion to change the indentation there. Not my finest moment.

**Review by lhearachel — APPROVED** [review]
Pre-emptively approving this pending consensus on function name paradigms.

**Comment by lhearachel** [comment]
We reached consensus on the naming scheme for functions moving forward. Please prefix your documented functions in this PR with appropriate verbage, per the naming scheme: `[<Module>_]<Imperative><Object>`

**Review by lhearachel — APPROVED** [review]

**Pull request merged**
